source/Octopus.Tentacle/Commands/RunAgentCommand.cs

```diff
@@ -46,7 +46,7 @@ public class RunAgentCommand : AbstractStandardCommand
             IWindowsLocalAdminRightsChecker windowsLocalAdminRightsChecker,
             AppVersion appVersion,
             ILogFileOnlyLogger logFileOnlyLogger,
-            IWorkspaceCleanerTask workspaceCleanerTask) : base(selector, log, logFileOnlyLogger)
+            Lazy<IWorkspaceCleanerTask> workspaceCleanerTask) : base(selector, log, logFileOnlyLogger)
```

Comment:
Nit: Like we discussed yesterday, I think it's worth putting a comment here stating why we do the `Lazy` thing (for future travellers 😁)

Also, do we think it's worth grouping all the `Lazy` dependencies together? It would make the comment more effective. I don't think it will have any flow-on affects 🤔 I'll leave the choice to you 😁

Reply:
I had created a separate story (sc-61799) to have another discussion and look around tidying up the Lazy dependencies. A comment is definitely good though, for the time being 👍